cmd/boost/deal_cmd.go

```diff
@@ -0,0 +1,343 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/filecoin-project/boost/storagemarket/types"
+	"github.com/mitchellh/go-homedir"
+
+	types2 "github.com/filecoin-project/boost/transport/types"
+	"github.com/filecoin-project/go-address"
+	cborutil "github.com/filecoin-project/go-cbor-util"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/api"
+	chain_types "github.com/filecoin-project/lotus/chain/types"
+	lcli "github.com/filecoin-project/lotus/cli"
+	"github.com/filecoin-project/specs-actors/actors/builtin/market"
+	"github.com/google/uuid"
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/multiformats/go-multiaddr"
+	cli "github.com/urfave/cli/v2"
+
+	inet "github.com/libp2p/go-libp2p-core/network"
+)
+
+const DealProtocolv120 = "/fil/storage/mk/1.2.0"
+
+var dealCmd = &cli.Command{
+	Name:  "deal",
+	Usage: "Make an online deal with Boost",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "repo",
+			Usage: "repo directory for Boost client",
+			Value: "~/.boost-client",
+		},
+		&cli.StringFlag{
+			Name:     "http-url",
+			Usage:    "http url to CAR file",
+			Required: true,
+		},
+		&cli.StringSliceFlag{
+			Name:  "http-headers",
+			Usage: "http headers to be passed with the request",
+		},
+		&cli.StringFlag{
+			Name:     "provider",
+			Usage:    "provider address",
```

Comment:
Let's make it clear the `provider` parameters is a filecoin address, rather than a multiaddr. Maybe also a description with an example.